java/src/main/java/org/eclipse/ditto/client/configuration/WebSocketMessagingConfiguration.java

```diff
@@ -128,15 +131,25 @@ public boolean isReconnectEnabled() {
 
         @Override
         public MessagingConfiguration build() {
-            final URI wsEndpointUri = appendWsPath(this.endpointUri, jsonSchemaVersion);
+            final URI wsEndpointUri= appendWsPathIfNecessary(this.endpointUri, jsonSchemaVersion);
             return new WebSocketMessagingConfiguration(jsonSchemaVersion, wsEndpointUri, reconnectEnabled,
                     proxyConfiguration, trustStoreConfiguration);
         }
 
-        private static URI appendWsPath(final URI baseUri, final JsonSchemaVersion schemaVersion) {
-            final String pathWithoutTrailingSlashes = baseUri.getPath().replaceFirst("/+$", "");
-            final String newPath = pathWithoutTrailingSlashes + WS_PATH + schemaVersion.toString();
-            return baseUri.resolve(newPath);
+        private static URI appendWsPathIfNecessary(final URI baseUri, final JsonSchemaVersion schemaVersion) {
+            if (needToAppendWsPath(baseUri, schemaVersion)) {
+                final String pathWithoutTrailingSlashes = baseUri.getPath().replaceFirst("/+$", "");
+                final String newPath = pathWithoutTrailingSlashes + WS_PATH + schemaVersion.toString();
+                return baseUri.resolve(newPath);
+            } else {
+                return baseUri;
+            }
+        }
+
+        private static boolean needToAppendWsPath(final URI baseUri, final JsonSchemaVersion schemaVersion) {
```

Comment:
Hm, what happens if the configured `baseUri` is configured to `/ws/1` but the schemaVersion is set to V_2?
Then I guess the URI could be: `/ws/1/ws/2`?

Reply:
Good point. I adapted the regex to check if the ws path is already appended.